common.tasks

```diff
@@ -4,28 +4,40 @@
   <UsingTask TaskName="Zip" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v4.0.dll">
     <ParameterGroup>
       <InputFileNames ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
+      <InputBaseDirectory ParameterType="System.String" Required="false" />
       <OutputFileName ParameterType="System.String" Required="true" />
       <OverwriteExistingFile ParameterType="System.Boolean" Required="false" />
-	</ParameterGroup>
+    </ParameterGroup>
     <Task>
       <Reference Include="System.IO.Compression" />
       <Using Namespace="System.IO.Compression" />
       <Code Type="Fragment" Language="cs">
-      <![CDATA[        
+      <![CDATA[
+        Log.LogMessage(MessageImportance.High, "=== Zipping files ===");
+        
         var fileMode = OverwriteExistingFile ? FileMode.Create : FileMode.CreateNew;
-        using (var archive = new ZipArchive(new FileStream(OutputFileName, fileMode), ZipArchiveMode.Create))
+        var baseDir  = string.IsNullOrWhiteSpace(InputBaseDirectory) ? string.Empty : Path.GetFullPath(InputBaseDirectory);
+        var filesToZip = InputFileNames.Select(f => Path.GetFullPath(f.ItemSpec)).Where(n => n.StartsWith(baseDir));
```

Comment:
Why would the user provided a collection containing files they don't want to zip? The idea is that `InputFileNames` contains only files that they user wants to zip. If they don't want to zip a file it should not be in the collection. 
The target is a basic building block and I am afraid we are now trying to be too smart. The result is that the user now needs to know what is happening inside the target because it is hard to tell how it behaves by just looking at parameters.